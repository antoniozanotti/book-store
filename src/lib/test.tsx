import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
