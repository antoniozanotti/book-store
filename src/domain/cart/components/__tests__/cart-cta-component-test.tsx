import { render } from "@/lib/test";
import { CartCtaComponent } from "../cart-cta-component";

describe("CartCtaComponent", () => {
  it("should render", async () => {
    const { container } = render(<CartCtaComponent />);
    expect(container).toMatchSnapshot();
  });
});
