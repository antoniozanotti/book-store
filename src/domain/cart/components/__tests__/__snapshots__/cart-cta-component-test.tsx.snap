// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartCtaComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <a
      class="flex items-center gap-2 pl-3 pr-3 lg:pr-0"
      href="/cart"
    >
      <svg
        class="lucide lucide-shopping-cart "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="21"
          r="1"
        />
        <circle
          cx="19"
          cy="21"
          r="1"
        />
        <path
          d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
        />
      </svg>
      <span>
        Cart
      </span>
    </a>
  </div>
</div>
`;
